File: code_samples/java/FunctionalCurrying.java
/**
 * Functional: Currying
 * AI/ML Training Sample
 */
public class FunctionalCurrying {
    
    private String data;
    
    public FunctionalCurrying() {
        this.data = "";
    }
    
    public void process(String input) {
        this.data = input;
    }
    
    public String getData() {
        return this.data;
    }
    
    public boolean validate() {
        return this.data != null && !this.data.isEmpty();
    }
    
    public static void main(String[] args) {
        FunctionalCurrying instance = new FunctionalCurrying();
        instance.process("example");
        System.out.println("Data: " + instance.getData());
        System.out.println("Valid: " + instance.validate());
    }
}

File: code_samples/rust/functional_currying.rs
//! Functional: Currying
//! AI/ML Training Sample

pub struct Currying {
    data: String,
}

impl Currying {
    pub fn new() -> Self {
        Self {
            data: String::new(),
        }
    }
    
    pub fn process(&mut self, input: &str) {
        self.data = input.to_string();
    }
    
    pub fn get_data(&self) -> &str {
        &self.data
    }
    
    pub fn validate(&self) -> bool {
        !self.data.is_empty()
    }
}

fn main() {
    let mut instance = Currying::new();
    instance.process("example");
    println!("Data: {}", instance.get_data());
    println!("Valid: {}", instance.validate());
}
